build: fix `stat` command on MacOS

slaperche-scality opened the pull request:
This should fix the issue on MacOS.

How to test:
- If you run `./doit.sh list` once you'll see that it installs a virtualenv first, then you see the output of `doit`.
- If you run `./doit.sh list` you should only see the output of `doit` (the venv is reused)
- If you `touch buildchain/requirements-Darwin.txt` (to fake an update of the dependencies) and then  `./doit.sh list` you should see like the first time (venv is reinstalled and then `doit` output).

slaperche-scality committed:
build: fix `stat` command on MacOS

On MacOS, the correct option is `-f %m` instead of `-c %Y`.

Refs: #841
Signed-off-by: Sylvain Laperche <sylvain.laperche@scality.com>

bert-e commented:
# Hello slaperche-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/841-fix-doit-sh-on-mac` into
`development/2.0`
* [w/2.1/bugfix/841-fix-doit-sh-on-mac](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/841-fix-doit-sh-on-mac) will be merged into `development/2.1` 
* [w/2.2/bugfix/841-fix-doit-sh-on-mac](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/841-fix-doit-sh-on-mac) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

bert-e commented:
# History mismatch



Merge commit #85046157374899514a26d7d332c04cac7cf87bc4 on the integration branch
`w/2.1/bugfix/841-fix-doit-sh-on-mac` is merging a branch which is neither the current
branch `bugfix/841-fix-doit-sh-on-mac` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/841-fix-doit-sh-on-mac` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

slaperche-scality committed:
build: fix dependencies for MacOS

`pip-compile` freeze the dependencies, but it doesn't handle the
platform-specific dependencies (it will uses the ones used by the
platform used to run `pip-compile`).

Consequentely, a `requirements.txt` generated on Linux may not work on
MacOS.

To avoid this issue (and to avoid having the user to run `pip-compile`
themselves) we now generate one `requirements.txt` per platform and let
`doit.sh` pick the right one (we already need to detect MacOS for the
`stat` command anyway).

Closes: #841
Signed-off-by: Sylvain Laperche <sylvain.laperche@scality.com>

slaperche-scality committed:
build: fix lint-python in tox

When linting Python code we also need to select the proper
requirements.txt, to do so we rely on an environment variable (OSTYPE).

Since we now depends on this environment variable, we need to pass it
from doit when calling `tox -e lint-python` and this can be done by
using a `CmdAction` (which only works with string, hence the
join/map/quote dance).

Refs: #842
Signed-off-by: Sylvain Laperche <sylvain.laperche@scality.com>

MonPote commented:
It works like a charm :)

slaperche-scality commented:
@bert-e approve

bert-e commented:
# History mismatch



Merge commit #85046157374899514a26d7d332c04cac7cf87bc4 on the integration branch
`w/2.1/bugfix/841-fix-doit-sh-on-mac` is merging a branch which is neither the current
branch `bugfix/841-fix-doit-sh-on-mac` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/841-fix-doit-sh-on-mac` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

slaperche-scality commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/841-fix-doit-sh-on-mac` into
`development/2.0`
* [w/2.1/bugfix/841-fix-doit-sh-on-mac](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/841-fix-doit-sh-on-mac) will be merged into `development/2.1` 
* [w/2.2/bugfix/841-fix-doit-sh-on-mac](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/841-fix-doit-sh-on-mac) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/5849e5f581860b34e3123341db1773a0cc68c4b8) did not succeed in branch w/2.2/bugfix/841-fix-doit-sh-on-mac.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

NicolasT commented on buildchain/requirements-Darwin.txt:
```diff
@@ -0,0 +1,9 @@
+#
```
Ugh :-( How is one with a Linux machine supposed to update this file now?

slaperche-scality replied:
Yup, Mac people are responsible to update it if they want it to work for them (I asked Carlito to generate this file) :S

I'm not that happy with the current solution but I didn't find a better way with `pip-compile` :-(
What I explored was:
- let people run `pip-compile` themselves: not good, they have to install `pip-tool` (an extra-dep, and extra-step) and we lose the reproducible build feature.
- using environment markers (as supported by pip): can't make it work with `pip-compile` though (it evaluate them at "compile" time instead of copying them to the generated `requirements.txt` so "compiling" on Linux force the use of `inotify` => issue is not solved).

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye slaperche-scality.
